main.go

```diff
@@ -121,3 +137,29 @@ func main() {
 		}
 	}
 }
+
+// reportExpiryMetrics takes a channel of VaultEvents, and reports expiry metrics on every successful renewal event.
+func reportExpiryMetrics(updates chan VaultEvent) {
+	for {
+		select {
+		case event := <-updates:
+			if event.Type == EventTypeFailure {
+				continue
+			}
+
+			certExpirationJson, ok := event.Secret["expiration"].(json.Number)
```

Comment:
These are not necessarily certs, so this var naming is a little off. Everything has a TTL